sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,157 @@
+|     task_info     |  struct  | Including the identification of orders and tasks, a task can be represented by a series of points and segments |
+|    battery_status    |  struct  | Including information such as remaining battery power, charging status, etc. |
+|   abnormal_events   | struct[] |         Record the exception information of the node         |
+|    robot_status     | boolean  |  indicates whether robot is ready for scheduling  |
+
+-  The fields contained in the task resource can be described as the table below:
+
+|          Field          |   Type   |                 Description                 |
+| :---------------------: | :------: | :-----------------------------------------: |
+|         orderID         |   uint   | The unique identification of the order in the system, used to identify whether multiple tasks belong to the same order |
+|         taskID          |   uint   | The orderID and taskID jointly identify the task status currently executed by the robot |
+|       allocated         |  string  | Empty if not assigned, otherwise the assigned robotID |
+|  point_status_sequence  | struct[] |     Sequence of points already received and not executed     |
+| segment_status_sequence | struct[] |    Sequence of segment already received and not executed     |
+| task_destination_ID  |  uint  | Describe the task destination point ID  |
+
+- The fields contained in the robotSync resource can be describe as follow:
+
+|          Field          |   Type   |             Description         |
+| :---------------------: | :------: | :-----------------------------: |
+|     registed_robots     | string[] | Keep a list of registered nodes, including nodes name and a list of node devices |
+
+## Controller design
+
+The Controller is mainly used to monitor the newly added custom resources in the cluster, or the changes of existing resources, and execute corresponding logic based on these informations.
+
+For *taskController*
+
+1. When a new task resource is created in the cluster, taskController should monitor this change
+2. The taskController match the task to a node that can execute the task
+3. Then the Controller should modify the task_info field of the node
+
+For the *robotControlelr*
+
+1. The robotController monitors the changes of the robot resources
+2. According to the changes, the Controller send the corresponding command to the node
+
+For the *registrationController*
+
+1. Listen for registration messages from robot nodes, and monitor the status of node at the same time
+2. Creating the robot CR if it receives the registration message and maintain the informations of robotSync CR. If the node goes offline, set the DeletionTimeStamp for the corresponding robot CR
+
+The processing logic of each Controller is shown in the figure below:
+
+<img src="./images/controllers.png" alt="controllers" style="zoom:80%;" />
+
+### Robot Registration
+
+<img src="./images/registration.png" alt="registration" style="zoom:90%;" />
+
```

Comment:
just for reference:

![image](https://github.com/kubeedge/community/assets/82074241/7d37888f-f4cd-4fc9-88be-dc3cdca10d8d)